File: AI.Core/OperatorsImplimentation/BasicOperators/Multiplication.cs
using AI.Core.ChannelsImplimentation;

namespace AI.Core.OperatorsImplimentation.BasicOperators
{
    public class Multiplication : Operator
    {
        public Multiplication(Entity thisEntity)
            : base(thisEntity)
        {
            EnterContactsCount = 2;
        }

        public override void Action()
        {
            var value = EnterContacts[0].Value * EnterContacts[1].Value;
            ExitContacts.SetValue(value);
        }
    }
}

File: NativeCore/OperatorsImplimentation/BasicOperators/Multiplication.cpp
#include "Multiplication.h"

using namespace AI::Core::ChannelsImplimentation;
namespace AI
{
	namespace Core
	{
		namespace OperatorsImplimentation
		{
			namespace BasicOperators
			{

				Multiplication::Multiplication(Entity *thisEntity) : Operator(thisEntity)
				{
					setEnterContactsCount(2);
				}

				void Multiplication::Action()
				{
					auto value = EnterContacts[0]->Value * EnterContacts[1]->Value;
					ExitContacts->SetValue(value);
				}
			}
		}
	}
}
